### PR TITLE
[Remove Stream/Optional usage] LPS-171834 headless PART 2 | 1-4

### PR DESCRIPTION
**Pull request by JanisZhang**
Add null-check for arrays. Fix Test. Thanks!

previous: https://github.com/liferay-core-infra/liferay-portal/pull/2913

### CODE REVIEW

File: modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/DDMFormValuesUtil.java [hunk 1]
```diff
@@ -163,34 +163,29 @@ private static Locale _getDefaultLocale(
 	}
 
 	private static Map<String, List<ContentField>> _toContentFieldsMap(
-		List<ContentField> contentFields) {
+		ContentField[] contentFields) {
 
-		if (contentFields.isEmpty()) {
+		if (contentFields == null) {
 			return new HashMap<>();
```
Comment (dantewang) [inline modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/DDMFormValuesUtil.java:169]: @JanisZhang I think you can use `Collections.emptyMap()` here..

Comment (JanisZhang) [reply modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/DDMFormValuesUtil.java:169]: @dantewang Then it need not create a separate Map object for each call. Thanks

**Comment by JanisZhang** [comment]
ci:stop

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test:headless

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by dantewang** [comment]
@JanisZhang See https://github.com/liferay-core-infra/liferay-portal/pull/2952, SF as per BChan's commit.

**Pull request closed without merging**